source/App/source/FunctionApp/Middleware/UserMiddleware.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.IdentityModel.Tokens.Jwt;
+using System.Net;
+using System.Security.Claims;
+using Energinet.DataHub.Core.App.Common.Abstractions.Users;
+using Energinet.DataHub.Core.App.Common.Users;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.Functions.Worker.Middleware;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Energinet.DataHub.Core.App.FunctionApp.Middleware;
+
+/// <summary>
+/// If possible, retrieved JWT from header and creates a user, which is then
+/// added to the function context and thereby made available in the executing function.
+/// </summary>
+public class UserMiddleware<TUser> : IFunctionsWorkerMiddleware
+    where TUser : class
+{
+    private const string MultiTenancyClaim = "multitenancy";
+
+    // DO NOT inject scoped services in the middleware constructor.
+    // DO use scoped services in middleware by retrieving them from 'FunctionContext.InstanceServices'
+    // DO NOT store scoped services in fields or properties of the middleware object. See https://github.com/Azure/azure-functions-dotnet-worker/issues/1327#issuecomment-1434408603
+    public UserMiddleware()
+    {
+    }
+
+    public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
+    {
+        var httpRequestData = await context.GetHttpRequestDataAsync().ConfigureAwait(false);
```

Comment:
```suggestion
        var httpRequestData = await context.GetHttpRequestDataAsync().ConfigureAwait(false);
```
What happens if "httpRequestData" is null?
Would it be preferable to call: next(context).ConfigureAwait(false) or throw an exception?

Reply:
We decided to throw an exception

Reply:
Fixed